apps/block_scout_web/assets/js/lib/analytics.js

```diff
@@ -0,0 +1,101 @@
+import { createStore } from './redux_helpers.js'
+
+import $ from 'jquery'
+import Analytics from 'analytics'
+import segmentPlugin from '@analytics/segment'
+import omit from 'lodash/omit'
+import uniqid from 'uniqid'
+
+let analytics
+let store
+
+const initialState = {
+  userID: localStorage.getItem('userID')
+}
+
+function reducer (state = initialState, action) {
+  switch (action.type) {
+    case 'PAGE_LOAD':
+    case 'ELEMENTS_LOAD': {
+      return Object.assign({}, state, omit(action, 'type'))
+    }
+    case 'SET_USER_ID': {
+      const id = uniqid()
+      localStorage.setItem('userID', id)
+      return Object.assign({}, state, { userID: id })
+    }
+    default:
+      return state
+  }
+}
+
+function trackEvents () {
+  $(function () {
+    // Page navigation
+    window.addEventListener('locationchange', function () {
+      analytics.page()
+    })
+
+    // Search box click
+    $('[data-selector="search-bar"]').on('click', function () {
+      analytics.track('Search bar clicked')
```

Comment:
Is it a legit event name? Could you double check with JB if it's not going to cause any issues, perhaps we should use target names from his document.